src/main/java/com/suse/saltstack/netapi/client/SaltStackJdkConnection.java

```diff
@@ -0,0 +1,110 @@
+package com.suse.saltstack.netapi.client;
+
+import com.suse.saltstack.netapi.config.SaltStackClientConfig;
+import com.suse.saltstack.netapi.exception.SaltStackException;
+import com.suse.saltstack.netapi.utils.SaltStackClientUtils;
+
+import com.google.gson.Gson;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+
+/**
+ * Class representation of a connection to SaltStack for issuing API requests using JDK's HttpURLConnection.
+ */
+public class SaltStackJdkConnection implements SaltStackConnection {
```

Comment:
I will follow your suggestion and focus on HttpComponents implementation, so this class will probably be removed.